pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -1,146 +1,91 @@
 { lib
-, stdenv
-, python
-, cmake
-, meson
-, vim
-, ruby
-, which
+
+  # nixpkgs functions
+, buildGoModule
+, buildVimPluginFrom2Nix
 , fetchFromGitHub
-, fetchurl
 , fetchpatch
-, llvmPackages
-, rustPlatform
-, buildGoModule
-, pkg-config
-, curl
-, openssl
-, libgit2
-, libiconv
-, xkb-switch
-, fzf
-, skim
-, stylish-haskell
+, fetchurl
+, stdenv
+, substituteAll
+
+  # Language dependencies
+, python
 , python3
+, rustPlatform
+
+  # Misc dependencies
+, Cocoa # vim-addon-manager
+, code-minimap # minimap-vim
+, dasht # vim-dasht
+, direnv # direnv-vim
+, fzf # fzfWrapper
+, gnome3 # vCoolor-vim
+, khard # deoplete-khard
+, languagetool # lens-vim, vim-grammarous
+, llvmPackages # clang_complete, clighter8
+, meson # meson
+, nim # fruzzy
+, nodePackages # coc-*
+, skim # skim, skim-vim
+, sqlite # sql-nvim
+, stylish-haskell # vim-stylish-haskell
+, tabnine # compe-tabine, completion-tabnine
+, vim # unicode-vim, vim-hier
+, which # vimproc-vim
+, xkb-switch # vim-xkbswitch
+, ycmd # YouCompleteMe
+
+  # command-t dependencies
+, rake
+, ruby
+
+  # cpsm dependencies
 , boost
+, cmake
 , icu
 , ncurses
-, ycmd
-, rake
-, gobject-introspection
-, glib
-, wrapGAppsHook
-, substituteAll
-, languagetool
-, tabnine
 
-, Cocoa
+  # LanguageClient-neovim dependencies
 , CoreFoundation
 , CoreServices
 
-, buildVimPluginFrom2Nix
-, nodePackages
-, dasht
-, sqlite
-, code-minimap
+  # sved dependencies
+, glib
+, gobject-introspection
+, wrapGAppsHook
 
-  # deoplete-khard dependency
-, khard
+  # vim-clap dependencies
+, curl
+, libgit2
+, libiconv
+, openssl
+, pkg-config
 
-  # vim-go dependencies
+# vim-go dependencies
 , asmfmt
 , delve
 , errcheck
+, gnused
+, go-motion
+, go-tools
+, gocode
+, gocode-gomod
 , godef
+, gogetdoc
+, golangci-lint
 , golint
 , gomodifytags
+, gopls
 , gotags
 , gotools
-, go-motion
-, gnused
-, reftools
-, gogetdoc
-, golangci-lint
-, impl
 , iferr
-, gocode
-, gocode-gomod
-, go-tools
-, gopls
-
-  # direnv-vim dependencies
-, direnv
-
-  # vCoolor dependency
-, gnome3
-
-  # fruzzy dependency
-, nim
+, impl
+, reftools
 }:
 
 self: super: {
 
-  vim2nix = buildVimPluginFrom2Nix {
```

Comment:
I am not feeling like proof reading all of this right now to spot any mistakes.

Reply:
Cool, thanks for the quick feedback on the arguments. 

FWIW, all the changes from here down are just me sorting the top-level attributes alphabetically (case-insensitive). So I don't think there's much to proofread.